fix: search and formatting

axs03 opened the pull request:
- Disabled disappearing categories when searching
``` javascript
// DISABLED: Clear categories container and show search results
document.getElementById("categories-container").innerHTML = "";
```

- Fixed bug in decrement button
``` javascript
// Prevent submit button showing up if currentQuantity is 0
if (currentQuantity > 0) {
  const newQuantity = Math.max(0, currentQuantity - 1)
  quantitySpan.textContent = newQuantity
  updateItemQuantity(item, location, area, item.category || "", newQuantity)
}
```

- Changed search bar styling for mobile and tablet

ssddOnTop commented:
@axs03 Can we remove the `Please enter at least 2 characters to search` restriction? 

Maybe we can use [fuse.js](https://www.fusejs.io/) for fuzzy search. We can listen to events of text change for the search bar.

So maybe we won't need to hide it anymore and we can remove the "Search" and "Clear" Buttons.

ssddOnTop reviewed (changes requested):
@axs03 Can we remove the `Please enter at least 2 characters to search` restriction? 

Maybe we can use [fuse.js](https://www.fusejs.io/) for fuzzy search. We can listen to events of text change for the search bar.

So maybe we won't need to hide it anymore and we can remove the "Search" and "Clear" Buttons.

axs03 commented:
@ssddOnTop 
- Removed <code>Please enter at least 2 characters to search</code> restriction
- Implemented event listener for search bar, using fuse for fuzzy search
- Removed search and clear buttons in correlation to event listener
- Fixed categories-container not clearing when dropdown location was changed
